### PR TITLE
Add first-class support for documented environment variables.

### PR DESCRIPTION
**Pull request by riannucci**
This allows applications to declare, document, and add default values for environment variables that they're sensitive too.

Includes support for the recently-added "advanced" feature to allow documentation of esoteric environment variables without cluttering the help output.

### CODE REVIEW

**Comment by riannucci** [comment]
Coveralls is making me feel bad, so I'll add some tests.

**Comment by riannucci** [comment]
Yay I made it less bad 😛

File: subcommands.go [hunk 1]
```diff
@@ -143,19 +180,51 @@ Use "{{.Name}} help -advanced" to display all commands.{{end}}
 	}
 	hasAdvanced := false
 	for _, c := range allCmds {
+		if namLen := len(c.Name()); namLen > widestCmd {
```
Comment (maruel) [inline subcommands.go]: the length should only include the commands that are visible?

File: subcommands.go [hunk 2]
```diff
@@ -143,19 +180,51 @@ Use "{{.Name}} help -advanced" to display all commands.{{end}}
...
+	}
 	data := map[string]interface{}{
 		"Title":           a.GetTitle(),
 		"Name":            a.GetName(),
 		"Commands":        cmds,
+		"EnvVars":         envVars,
 		"ShowAdvancedTip": (hasAdvanced && !includeAdvanced),
 	}
-	tmpl(out, usageTemplate, data)
+	tmpl(out, fmt.Sprintf(usageTemplate, widestCmd+2, widestEnvVar+2), data)
```
Comment (maruel) [inline subcommands.go]: why +2 ? You can add two empty space instead

**Review by maruel — CHANGES_REQUESTED** [review]
Only small nits.

The main drawback is that it is a breaking change.

File: subcommands.go (hunk 2 above)
Comment (riannucci) [reply subcommands.go]: Derp

File: subcommands.go (hunk 1 above)
Comment (riannucci) [reply subcommands.go]: Yeah, refactored this a bit. It was confusing (my fault from previous change)

**Comment by riannucci** [comment]
OK, Addressed comments, PTAL

File: subcommands.go [hunk 3]
```diff
@@ -127,35 +159,68 @@ func usage(out io.Writer, a Application, includeAdvanced bool) {
...
+		}
+	}
+
+	type envVarEntry struct {
+		Name      string
+		ShortDesc string
+		Default   string
+	}
+	widestEnvVar := 0
+	envVars := []envVarEntry(nil)
```
Comment (maruel) [inline subcommands.go:195]: I generally prefer the traditional form for this but I don't mind much.

**Review by maruel — APPROVED** [review]

**Pull request merged**